### PR TITLE
WT-8361 Increase assert limit in search near 01 cppsuite test

### PR DESCRIPTION
**Pull request by jiechenbo**
The fix involved in the PR, regards to increasing the assert limit for search near 01 cppsuite test. The reason for this change is because we previously didn't account if a thread would perform multiple search nears before another thread completing one. To account for this fact, I have doubled the expected entries to account for this anomaly. Now this change presumes that it is not possible to have all threads performing two search nears which on theory is impossible.

### CODE REVIEW

File: test/cppsuite/tests/search_near_01.cxx [hunk 1]
```diff
@@ -237,14 +238,16 @@ class search_near_01 : public test_harness::test {
...
-                 * prefix search near. This is dependent on how many read threads are present in the
-                 * test. Account for this by creating a small buffer using thread count. Assert that
-                 * the number of expected entries is the upper limit which the prefix search near
-                 * can traverse and the prefix fast path is incremented.
+                 * Due to the concurrency of multiple threads and how WiredTiger increments the
+                 * entries skipped stat, it is possible that a thread can perform multiple search
+                 * nears before another can finish one. Account for this problem by creating a
+                 * buffer roughly equivalent to the calculated 2 * expected entries. Assert that the
+                 * number of expected entries is the upper limit which the prefix search near can
+                 * traverse and the prefix fast path is incremented.
```
Comment (etienneptl) [inline test/cppsuite/tests/search_near_01.cxx:246]: Add a note about the value `40` as well. You could also create a variable for it if possible.

Comment (jiechenbo) [reply test/cppsuite/tests/search_near_01.cxx]: done! let me know if you like the comment.

File: test/cppsuite/configs/search_near_01_stress.txt [hunk 2]
```diff
@@ -13,6 +13,7 @@ workload_generator=
     (
         collection_count=3,
         key_count_per_collection=100,
+        value_size=30
```
Comment (jiechenbo) [inline test/cppsuite/configs/search_near_01_stress.txt:16]: I added this. This aims to increase the stress testing.

**Review by etienneptl — APPROVED** [review]
LGTM, please create a follow-up ticket so we can rework this test and avoid this buffer when we check the stats.

**Comment by jiechenbo** [comment]
>LGTM, please create a follow-up ticket so we can rework this test and avoid this buffer when we check the stats.
I have created WT-8417 for the follow up ticket! 👍

**Review by AlexeyAnisimovAU — APPROVED** [review]
In my opinion, this test requires a redesign to make the verification more reliable.
I'm going to LGTM this fix because there is WT-8417 for a future redesign.

**Comment by jiechenbo** [comment]
evergreen retry

**Pull request merged**